backend/src/apiserver/resource/resource_manager.go

```diff
@@ -281,6 +399,11 @@ func (r *ResourceManager) CreateRun(apiRun *api.Run) (*model.RunDetail, error) {
 		}
 	}
 
+	err = r.dropUserGcpSaIfNotConfigured(workflow)
+	if err != nil {
+		return nil, util.NewInternalServerError(err, "Failed to verify whether GCP secret is configured")
```

Comment:
Ask for review. Call out that this is probably something that would be controversial. I discussed with @jingzhang36 about this. We concluded that because only users whose workflow contains `use_gcp_secret` can possibly hit this error. It's clearer if the error is returned to user, instead of silently swallowed (probably with a warning message that no user will ever see).